teacher.rb

```diff
@@ -0,0 +1,26 @@
+require './main'
+# Public Teacher class used for creating teacher objects
+#  Teacher class inherits from person class.
+#
+# All methods are class methods and should be called on the Person class.
+#
+# Examples
+#
+# teacher = Teacher.new('unknown')
+# teacher.can_use_services?
+# # => true
+class Teacher < Person
+  attr_reader :specialization
+
+  # Public: Initializes a new Teacher object.
+  # specialization - The String name of the specialization.
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```

Comment:
- [ ] A better way to initialize this class would be like this:

```RUBY
def initialize(age, specialization, name = "Unknown", parent_permission = true)
  super(age, name, parent_permission)
  @specialization = specialization
end

```

That way, `super` will send the required teacher arguments to the parent `initialize` method. 💪

Reply:
Done